src/lib/schema/SchemaEntry.ts

```diff
@@ -0,0 +1,202 @@
+
+export class SchemaEntry {
+
+	/**
+	 * The KlasaClient for this SchemaEntry.
+	 */
+	public client: Client | null = null;
+
+	/**
+	 * The schema that manages this instance.
+	 */
+	public readonly parent: Schema | SchemaFolder;
+
+	/**
+	 * The key of this entry relative to its parent.
+	 */
+	public readonly key: string;
+
+	/**
+	 * The absolute key of this entry.
+	 */
+	public readonly path: string;
+
+	/**
+	 * The type of data this entry manages.
+	 */
+	public type: string;
+
+	/**
+	 * Whether or not this entry should hold an array of data.
+	 */
+	public array: boolean;
+
+	/**
+	 * The default value this entry will set when reverting a setting back to or when the key was not set.
+	 */
+	public default: SerializableValue;
+
+	/**
+	 * The minimum value for this entry.
+	 */
+	public minimum: number | null;
+
+	/**
+	 * The maximum value for this entry.
+	 */
+	public maximum: number | null;
+
+	/**
+	 * Whether this entry should inclusively or exclusively check minimum and maximum on key validation.
+	 */
+	public inclusive: boolean;
+
+	/**
+	 * Whether or not this entry should be configurable by the configuration command.
+	 */
+	public configurable: boolean;
+
+	/**
+	 * The filter to use for this entry when resolving.
+	 */
+	public filter: SchemaEntryFilterFunction | null;
+
+	/**
+	 * Whether or not values managed by this entry should be resolved.
+	 */
+	public shouldResolve: boolean;
+
+	public constructor(parent: Schema | SchemaFolder, key: string, type: string, options: SchemaEntryOptions = {}) {
+		this.client = null;
+		this.parent = parent;
+		this.key = key;
+		this.path = this.parent.path.length === 0 ? this.key : `${this.parent.path}.${this.key}`;
+		this.type = type.toLowerCase();
+		this.array = typeof options.array === 'undefined' ? typeof options.default === 'undefined' ? false : Array.isArray(options.default) : options.array;
+		this.default = typeof options.default === 'undefined' ? this.generateDefaultValue() : options.default;
+		this.minimum = typeof options.minimum === 'undefined' ? null : options.minimum;
+		this.maximum = typeof options.maximum === 'undefined' ? null : options.maximum;
+		this.inclusive = typeof options.inclusive === 'undefined' ? false : options.inclusive;
+		this.configurable = typeof options.configurable === 'undefined' ? this.type !== 'any' : options.configurable;
+		this.filter = typeof options.filter === 'undefined' ? null : options.filter;
+		this.shouldResolve = typeof options.resolve === 'undefined' ? true : options.resolve;
+	}
+
+	/**
+	 * Get the serializer that manages this instance.
+	 */
+	public get serializer(): Serializer | null {
+		if (this.client === null) throw new Error('Cannot retrieve serializers from non-initialized SchemaEntry.');
+		return this.client.serializers.get(this.type) || null;
+	}
+
+	/**
+	 * Edits this SchemaEntry instance.
+	 * @param options The options to edit
+	 */
+	public edit(options: SchemaEntryEditOptions = {}): this {
```

Comment:
Question: What happens if I edit the schema after the bot goes ready? Will things break big time? Or are schema edits ignored afterwards? Will new settings instances have the new schema, thus causing issues?